### DISCUSSION TITLE
Cannot read data from SGP40 sensor

### DISCUSSION THREAD

**Discussion by adysanro** [micropython/micropython#9193] [Hardware & Peripherals] [answered]
Hi,
I am new to Python, boards, soldering and stuff like this. I am playing with a Raspberry Pi Pico W and a SGP40 air quality sensor I purchased from Aliexpress. 

Issue: cannot read sensor data. I am not getting a message or error message. The output section remains empty.

I ran an i2c.scan(). The device is detected from GP pins 12 and 13 at address 0x59 (which is the address mentioned in the datasheet for SGP40). I am using the same detected GP pins 12 and 13 in my sample Python code.
I am using the Agners library to read sensor data: https://github.com/agners/micropython-sgp40
I am able to read data from other sensors on the same board and on the same GPIO pins.
See the screenshot below and the code.zip file attached for more context.

What am I missing?

![2022-09-02 19_26_44-](https://user-images.githubusercontent.com/87180477/188197987-1f1f4fa6-4bd9-4a07-8bef-45fffafca945.png)

[code.zip](https://github.com/micropython/micropython/files/9479646/code.zip)

Thanks,
Adrian

**Comment by jimmo** [comment, marked as answer]
Your program doesn't ever get a chance to print anything

```py
while True:
    value = sgp40.measure_raw()
    time.sleep(1)

print(value)
```

The `print()` is not inside the loop, and because the loop never terminates, the `print()` will never execute.

I suspect you want to write instead:

```py
while True:
    value = sgp40.measure_raw()
    print(value)
    time.sleep(1)
```

**Reply by adysanro** [reply]
Aah, such a noob mistake :) It's giving data now.
PS: before I reached this "no output" issue I struggled with at least another 5 issues for half a day ranging from: bad soldering, not using the proper GP pins vs how they're declared in the code, an "OSError: [Errno 5] EIO" message and others. What can I say, it's quite a learning experience :)
